#!/usr/bin/python
PACKAGE = "ros_adas2019_examples"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kP", double_t, 0, "Controller parameter - proportional part", 15, 0, 100)
gen.add("kI", double_t, 0, "Controller parameter - integral part", 15, 0, 20)
gen.add("kD", double_t, 0, "Controller parameter - differential part", 0, 0, 10)
gen.add("minimumOutputPower", double_t, 0, "minimum output power (values below this setting get clamped to avoid beeping)", 0.3, 0, 1)
gen.add("maximumOutputPower", double_t, 0, "maximum output power (values above this setting get clamped to avoid burning the fuse)", 25, 0, 100)

exit(gen.generate(PACKAGE, "ros_adas2019_examples", "WheelSpeedController"))
